traitsui/wx/image_enum_editor.py

```diff
@@ -113,13 +117,24 @@ def init(self, parent):
         """ Finishes initializing the editor by creating the underlying toolkit
             widget.
         """
-        self._create_image_grid(parent)
+        super(CustomEditor, self).init(parent)
```

Comment:
The `parent` is not actually used by `BaseEnumEditor.init`. The base class's `init` does not do what the Editor Interface says it should do: "Finishes initializing the editor by creating the underlying widget".  In fact, all the subclasses rely on the base class's `init` not to create any widget at all.

This `BaseEnumEditor` can be a Mixin that does not subclass `Editor`, but it can still expect subclasses to be instances of `Editor`. The mixin can provide "setup" and "teardown" for subclasses to call in their `init` and `dispose` method. Then `BaseEnumEditor` can be completely free of the GUI toolkit and be refactored out.

This kind of refactoring is probably out-of-scope for this PR. Unless you feel very strongly against this idea or you'd like do the refactoring here, I will open a separate issue for this.

Reply:
Sounds good to me and I agree that it's best done separately

Reply:
Opened https://github.com/enthought/traitsui/issues/857